pyam/ops.py

```diff
@@ -0,0 +1,58 @@
+import pandas as pd
+
+from pyam.utils import _meta_idx, merge_meta
+
+
+def subtract(a, b):
+    return a - b
+
+
+class BinaryOp(object):
+
+    def __init__(self, a, b, ignore_meta_conflict=False):
+        self.a_df = a
+        self.b_df = b
+        
+        # Generate meta data merge if two dataframes are not the same,
+        # and do so early to fail early
+        # TODO: merge_meta needs tests
+        self.meta = pd.DataFrame() if a is b else \
```

Comment:
1) if this is an attempt to bypass merging `meta` in the `subtract_inplace()` function, this won't work because `a` and `b` are filtered for different values, hence different.

2) on a meta-level (pun intended), say you subtract scenario a from scenario b, what should happen to the `meta` table? New blank meta table, checking if they are equivalent and keeping them? Should something be written to the log?

Reply:
This is only hit if the dataframes are the exact same object (i.e., `df.subtract(df)`) - this was mostly carried over from previous implementations. I'm also happy just remove this - all tests (except for checking that its empty) pass. Let me know what you think!

Reply:
removed - as implemented the merge always happens. we can optimize later.

Reply:
1. why would you subtract two times the exact dataframe - that would be all zero, right?

2. if you do a within-dataframe operation, you can skip that (which you should do for speed and sanity)

3. The larger point of my comment remains - if you are doing an operation on variable, region or unit (does that even make sense?), one can easily do `merge_meta()`. But what should happen when there is a new model or scenario created?

Reply:
Agree on 1 and 2.

On 3 - great question. 

I would argue that operations with `axis in ['model', 'scenario']` should have undefined (no) metadata. By definition, metadata is keyed on these attributes. Any desired metadata on this `new_name` should be added by the user, right? 

E.g., in which temperature category bin does `scen1 - scen2` reside? Doesn't make sense, shouldn't be in the `meta` table. It would be super nice to be able to do

```
a = df.filter(scenario='baseline', keep=False)
b = df.filter(scenario='baseline')
result = a.subtract(b, new_name='difference from baseline')
result.bar_plot()
```

These kinds of operations are valuable (e.g., diff from a baseline), but don't need to add metadata. I believe we are in agreement here.